synthtool/languages/java.py

```diff
@@ -296,3 +297,33 @@ def bazel_library(
     )
 
     return library
+
+
+def common_templates(metadata: Dict = {}, excludes: List[str] = [], **kwargs) -> None:
+    """Generate common templates for a Java Library
+
+    Fetches information about the repository from the .repo-metadata.json file,
+    information about the latest artifact versions and copies the files into
+    their expected location.
+
+    Args:
+        metadata (Dict, optional): Additional template metadata
+        excludes (List[str], optional): List of template paths to ignore
+        **kwargs: Additional options for CommonTemplates.java_library()
+    """
+    repo_metadata = common.load_repo_metadata()
```

Comment:
It might be worth looking at how we approach some of the logic in the `nodejs` repo, if you have any similar needs around adding samples to READMEs (we do some fancy :jeans: loading of the samples directories).

Reply:
We don't have any samples to pull *yet* so I didn't want to refactor too much